welkin/models/base.py

```diff
@@ -20,11 +23,23 @@ def __getattr__(self, name):
 
 
 class Resource(dict, SchemaBase):
+    nested_objects = {}
+
     def __getattr__(self, name):
         try:
-            return super().__getitem__(name)
+            value = super().__getitem__(name)
         except KeyError:
-            return super().__getattr__(name)
+            value = super().__getattr__(name)
+
+        if name in self.nested_objects:
+            cls = getattr(sys.modules["welkin.models"], self.nested_objects[name])
```

Comment:
I love this! Not only does it work around the circular import issue, but it properly raises `AttributeError` if there's any sort of issue.